mirdata/track.py

```diff
@@ -33,11 +33,127 @@ def __repr__(self):
             if val.__doc__ is None:
                 raise ValueError("{} has no documentation".format(prop))
 
-            val_type_str = val.__doc__.split(':')[0]
+            val_type_str = val.__doc__.split(":")[0]
             repr_str += "  {}: {},\n".format(prop, val_type_str)
 
         repr_str += ")"
         return repr_str
 
     def to_jams(self):
         raise NotImplementedError
+
+
+class MultiTrack(Track):
+    """MultiTrack class.
+
+    A multitrack class is a collection of track objects and their associated audio
+    that can be mixed together.
+    A multitrack is iteslf a Track, and can have its own associated audio (such as
+    a mastered mix), its own metadata and its own annotations.
+
+    """
+
+    def _check_mixable(self):
+        if not hasattr(self, "tracks") or not hasattr(self, "track_audio_attribute"):
+            raise NotImplementedError(
+                "This MultiTrack has no tracks/track_audio_attribute. Cannot perform mixing"
+            )
+
+    def get_target(self, track_keys, weights=None, average=True, enforce_length=True):
+        """Get target which is a linear mixture of tracks
+
+        Args:
+            track_keys (list): list of track keys to mix together
+            weights (list or None): list of positive scalars to be used in the average
+            average (bool): if True, computes a weighted average of the tracks
+                if False, computes a weighted sum of the tracks
+            enforce_length (bool): If True, raises ValueError if the tracks are 
+                not the same length. If False, pads audio with zeros to match the length
+                of the longest track
+        
+        Returns:
+            target (np.ndarray): target audio with shape (n_channels, n_samples)
+
+        Raises:
+            ValueError: 
+                if sample rates of the tracks are not equal
+                if enforce_length=True and lengths are not equal
+
+        """
+        self._check_mixable()
+        signals = []
+        lengths = []
+        sample_rates = []
+        for k in track_keys:
+            audio, sample_rate = getattr(self.tracks[k], self.track_audio_attribute)()
```

Comment:
@rabitt I get an error at this line. why is the audio attribute a callable? audio is a property of a Track as far as I understood.
TypeError: 'tuple' object is not callable

this works:
`audio, sample_rate = getattr(self.tracks[k], self.track_audio_attribute)
`

Reply:
ah! yes you're right. Fixing